### PR TITLE
fix: grace period validation and parsing

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** The resubmission grace period wasn't being correctly parsed from xml or correctly validated in studio. 

JIRA: https://2u-internal.atlassian.net/browse/AU-2071

**What changed?**

- xml parsing should default grace period to empty string rather than none
- shouldn't be validating the grace period field when the feature is disabled

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by pomegranited** [comment]
Hi @jansenk , this change looks great, thank you for submitting it :)
I'm having issues with my devstack, but will do my best to get this tested and approved tomorrow or next week.

**Review by pomegranited — APPROVED** [review]
:+1: Thank you for fixing this issue so cleanly @jansenk , and for adding great tests.

- [x] I tested this on my devstack:
   * created a new ORA unit
   * played with enabling/disabling resubmissions and valid/invalid data in the grace period field
   * exported & re-imported the course with edited ORA2 OLX, to play with the above permutations.
   * checked that the UI still flags invalid values from the imported OLX and works as desired.
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] <strike>Includes documentation</strike> N/A
- [x] <strike>User-facing strings are extracted for translation</strike> N/A

**Review by mariajgrimaldi — COMMENTED** [review]
Oh, I totally missed this. Thank you folks!
